index.js

```diff
@@ -1,14 +1,42 @@
 // These are all the exports for the NodeJS environment.
 // The idea is that JointJS can be used exactly the same way in the browser and also in Node.
 
-module.exports = {
+var Backbone = require('backbone');
+Backbone.$ = require('./lib/jquery');
 
-    dia: {
-        Graph: require('./src/joint.dia.graph').Graph,
-        Link: require('./src/joint.dia.link').Link,
-        Element: require('./src/joint.dia.element').Element
-    },
-    util: require('./src/core').util,
-    shapes: require('./plugins/shapes'),
-    alg: require('./plugins/alg/alg')
+module.exports = {
+  util:       {},
+  shapes:     {},
+  connectors: {},
+  routers: {},
+  alg:        {},
+  dia:    {
+    Cell:     {},
+    CellView: {},
+    Graph:    {},
+    Link:     {},
+    Element:  {},
+    Paper:    {}
+  }
 };
+var jointjs = require('jointjs');
```

Comment:
Is this actually working for you? Doesn't seem to me that you are exporting the `jointjs` library here (but the namespaces only).